example/pytorch.md

```diff
@@ -0,0 +1,143 @@
+-->
+
+
+# PyTorch on PAI
+
+This guide introduces how to run [PyTorch](http://pytorch.org/) workload on PAI.
+The following contents show some basic PyTorch examples, other customized PyTorch code can be run similarly.
+
+
+## Contents
+
+1. [Basic environment](#basic-environment)
+2. [Advanced environment](#advanced-environment)
+3. [PyTorch examples](#pytorch-examples)
+4. [Frequently asked questions](#faq)
+
+
+## Basic environment
+
+First of all, PAI runs all jobs in Docker container.
+[Install Docker-CE](https://docs.docker.com/install/linux/docker-ce/ubuntu/) if you haven't. Register an account at public Docker registry [Docker Hub](https://hub.docker.com/) if you do not have a private Docker registry.
+
+We need to build a PyTorch image with GPU support to run PyTorch workload on PAI, this can be done in two steps:
+
+1. Build a base Docker image for PAI. We prepared a [Dockerfile](../job-tutorial/Dockerfiles/cuda8.0-cudnn6/Dockerfile.build.base) which can be built directly.
+
+    ```bash
+    $ cd ../job-tutorial/Dockerfiles/cuda8.0-cudnn6
+    $ sudo docker build -f Dockerfile.build.base \
+    >                   -t pai.build.base:hadoop2.7.2-cuda8.0-cudnn6-devel-ubuntu16.04 .
+    $ cd -
+    ```
+
+2. Prepare PyTorch envoriment in a Dockerfile using the base image.
+
+    Write a PyTorch Dockerfile and save it to `Dockerfile.example.pytorch`:
+
+    ```dockerfile
+    FROM pai.build.base:hadoop2.7.2-cuda8.0-cudnn6-devel-ubuntu16.04
+
+    # install PyTorch dependeces using pip
+    RUN pip install torch torchvision
+
+    # clone PyTorch examples
+    RUN git clone https://github.com/pytorch/examples.git
+    ```
+
+    Build the Docker image from `Dockerfile.example.pytorch`:
+
+    ```bash
+    $ sudo docker build -f Dockerfile.example.pytorch -t USER/pai.example.pytorch .
+    ```
+
+    Push the Docker image to a Docker registry:
+
+    ```bash
+    $ sudo docker push USER/pai.example.pytorch
+    ```
+    *Note: Replace USER with the Docker Hub username you registered, you will be required to login before pushing Docker image.*
+
+
+## Advanced environment
+
+You can customize runtime PyTorch environment in `Dockerfile.example.pytorch`, for example, adding other dependeces in Dockerfile:
+
+```dockerfile
+FROM pai.build.base:hadoop2.7.2-cuda8.0-cudnn6-devel-ubuntu16.04
+
+# install other packages using apt-get
+RUN apt-get -y update && apt-get -y install PACKAGE
+
+# install other packages using pip
+RUN pip install torch torchvision PACKAGE
+
+# clone PyTorch examples
+RUN git clone https://github.com/pytorch/examples.git
+```
+
+
+# PyTorch examples
+
+To run PyTorch examples in PAI, you need to prepare a job configuration file and submit it through webportal.
+Here're some configuration file examples:
+
+### [mnist](https://github.com/pytorch/examples/tree/master/mnist)
+```json
+{
+  "jobName": "pytorch-mnist",
+  "image": "USER/pai.example.pytorch",
```

Comment:
>USER/pai.example.pytorch [](start = 12, length = 24)

can you publish an example pytorch image in dockerhub. so user can run a pytorch job directly in their own PAI cluster.

Reply:
published at [Docker Hub](https://hub.docker.com/r/paiexample/pai.example.pytorch/)